source/FreeplayState.hx

```diff
@@ -39,7 +39,8 @@ class FreeplayState extends MusicBeatState
 
 		for (i in 0...initSonglist.length)
 		{
-			songs.push(new SongMetadata(initSonglist[i], 1, 'gf'));
+			var data:Array<String> = initSonglist[i].split(':');
```

Comment:
It would be good to have some basic error checking and logging so mistakes in freeplaySonglist.txt don't crash the game or get ignored.